File: src/boj/B17609_회문.java
package boj;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class B17609_회문 {
    static StringBuffer sb = new StringBuffer();

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(bf.readLine());

        for(int i=0; i<T; i++){
            String input = bf.readLine();

            int start = 0;
            int end = input.length() - 1;
            int ans = 0;

            while(start < end){
                if(input.charAt(start) == input.charAt(end)){
                    start++;
                    end--;
                }
                else if(check2(input.substring(start, end))){ // start ~ end - 1
                    ans = 1;
                    break;
                }
                else if(check2(input.substring(start + 1, end + 1))) { // start + 1 ~ end
                    ans = 1;
                    break;
                }
                else {
                    ans = 2;
                    break;
                }
            }
            sb.append(ans + "\n");
        }

        System.out.println(sb.toString());
    }

    static boolean check2(String s){
        int start = 0;
        int end = s.length() - 1;

        while(start<end){
            if(s.charAt(start) == s.charAt(end)){
                start++;
                end--;
            }
            else return false;
        }
        return true;
    }
}

File: src/boj/B17609_회문.py
t = int(input())
ans = ''
for i in range(t):
    s = input()
    if s[::]==s[::-1]: # 회문인지 판단
        ans += '0'+'\n'
    else: # 유사회문인지, 회문이 아닌지 판단
        x = 0
        y = len(s)-1
        n = 0
        while x < y:
            if s[x] != s[y]:
                if s[x+1:y+1] == s[x+1:y+1][::-1] or s[x:y] == s[x:y][::-1]:
                    ans += '1' + '\n'
                    break
                else:
                    ans += '2' + '\n'
                    break
            x += 1
            y -= 1
print(ans)

